tests/foreman/cli/test_logging.py

```diff
@@ -0,0 +1,251 @@
+"""CLI tests for logging.
+
+:Requirement: Logging
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: logging
+
+:TestType: Functional
+
+:CaseImportance: Medium
+
+:Upstream: No
+"""
+import re
+from fauxfactory import gen_string
+from robottelo import ssh
+from robottelo.ssh import upload_file
+from robottelo import manifests
+from robottelo.test import CLITestCase
+from robottelo.cli.factory import (
+    make_org,
+    make_product_wait,
+    make_repository,
+)
+from robottelo.cli.subscription import Subscription
+
+
+class SimpleLoggingTestCase(CLITestCase):
+    """Test class for default logging to files."""
+
+    org = None
+    product = None
+
+    @classmethod
+    def setUpClass(cls):
+        """Tests for logging to files"""
+        super(SimpleLoggingTestCase, cls).setUpClass()
+
+        # need own org for the manifest refresh test
+
+        cls.org = make_org(cached=True)
+        cls.product = make_product_wait(
+            {u'organization-id': cls.org['id']},
+            )
+
+    def _make_repository(self, options=None):
+        """Makes a new repository and asserts its success"""
+        if options is None:
+            options = {}
+
+        if not options.get('product-id'):
+            options[u'product-id'] = self.product['id']
+
+        return make_repository(options)
+
+    def test_positive_logging_from_foreman_core(self):
+        """Check that GET command to Hosts API is logged and has request ID.
+
+        :id: 0785260d-cb81-4351-a7cb-d7841335e2de
+
+        :expectedresults: line of log with GET has request ID
+
+        :CaseImportance: Medium
+        """
+
+        GET_line_found = False
+        source_log = '/var/log/foreman/production.log'
+        test_logfile = '/var/tmp/logfile_from_foreman_core'
+        with ssh.get_connection() as connection:
+            # get the number of lines in the log before the test
+            line_count_1 = connection.run('wc -l < {0}'.format(source_log),
+                                          output_format='plain')
+            self.assertEqual(line_count_1.return_code, 0, "BASH command error?")
+            # hammer command for this test
```

Comment:
above two lines are common all the test functions, good to be a candidate for reusable function in a class

Reply:
@omkarkhatavkar please see new function: "def line_count(self, file, connection=None)"
Line 33